api/turing/server/application.go

```diff
@@ -117,8 +118,18 @@ func Run() {
 	// HealthCheck Handler
 	AddHealthCheckHandler(r, "/v1/internal", db)
 
+	// Override the bundle file with user specified overrides and serve files
+	openAPISpecBytes, err := utils.MergeTwoYamls(
+		cfg.OpenapiConfig.SpecFile,
+		*cfg.OpenapiConfig.SpecOverrideFile,
```

Comment:
Can you implement the patching logic as a method of `config.OpenapiConfig` struct?  Something like this:
```go
func (c * OpenapiConfig) SpecData() ([]byte, error) {
  return utils.MergeTwoYamls(c.SpecFIle, c. SpecOverrideFile)
}
```

The power of abstractions helps you to manage the complexity of certain parts of your application by encapsulating the implementation details within the bounds of the abstraction. I.e. I likely don't want to know anything about the specfile or the patching logic of it at the root level of my application, but what I need is a laconic contract on how to retrieve my API specification.

Also, I'm not sure if passing `openAPISpecBytes` into `AddAPIRoutesHandler` is necessary as you already have an instance of `Config` passed into this method, meaning that you can retrieve API spec data from the `cfg.OpenapiConfig`

Reply:
Done